### PR TITLE
Add identities and directories for cdp

### PR DESCRIPTION
**Pull request by shirleylxie**

### CODE REVIEW

**Comment by shirleylxie** [comment]
Travis failed with error: "ERROR: More than one .egg-info directory found in /tmp/pip-pip-egg-info-o1xw8f4u".
It should be fixed when https://github.com/Isilon/isilon_hadoop_tools/pull/87 is merged.

**Comment by shirleylxie** [comment]
Test Done:
Create users:
`root@remotedev-sxie-rdu:/git/isilon_hadoop_tools/venv/bin# ./isilon_create_users --dist cdp --onefs-password a --zone System  --no-verify 10.205.230.227`
http://paste.west.isilon.com/247276

Create dirs:
`root@remotedev-sxie-rdu:/git/isilon_hadoop_tools/venv/bin# ./isilon_create_directories --dist cdp --onefs-password a --zone System  --no-verify 10.205.230.227`
http://paste.west.isilon.com/247278

File: src/isilon_hadoop_tools/directories.py [hunk 1]
```diff
@@ -132,6 +133,51 @@ def cdh_directories(identity_suffix=None):
     return directories
 
 
+def cdp_directories(identity_suffix=None):
+    """Directories needed for Cloudera Distribution including Hadoop"""
```
Comment (tucked) [inline src/isilon_hadoop_tools/directories.py:137]:
```suggestion
    """Directories needed for Cloudera Data Platform"""
```

File: src/isilon_hadoop_tools/identities.py [hunk 2]
```diff
@@ -387,6 +388,77 @@ def cdh_identities(zone):
         identities['users']['admin'] = ('admin', set())
     return identities
 
+def cdp_identities(zone):
+    """Identities needed for Cloudera Distribution including Hadoop"""
```
Comment (tucked) [inline src/isilon_hadoop_tools/identities.py:392]:
```suggestion
    """Identities needed for Cloudera Data Platform"""
```

File: src/isilon_hadoop_tools/identities.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
     # Functions
     'cdh_identities',
     'hdp_identities',
+    'cdp_identities',
```
Comment (tucked) [inline src/isilon_hadoop_tools/identities.py:16]: Let's keep these alphabetized.
```suggestion
    'cdp_identities',
    'hdp_identities',
```

**Review by tucked — COMMENTED** [review]
Looks like the CLI plumbing (e.g. `--dist cdp`) is missing? I thought you might add it later, but I don't think the commands you tested with would work without it. 🤔

**Comment by tucked** [comment]
We'll also need to update the tests to exercise the new code:
https://github.com/Isilon/isilon_hadoop_tools/blob/238bf718e6eb439a49aca8d8d5ae29df95b176a6/tests/conftest.py#L590
https://github.com/Isilon/isilon_hadoop_tools/blob/238bf718e6eb439a49aca8d8d5ae29df95b176a6/tests/conftest.py#L604-L607
https://github.com/Isilon/isilon_hadoop_tools/blob/238bf718e6eb439a49aca8d8d5ae29df95b176a6/tests/conftest.py#L615-L618
https://github.com/Isilon/isilon_hadoop_tools/blob/238bf718e6eb439a49aca8d8d5ae29df95b176a6/tests/test__scripts.py#L24
https://github.com/Isilon/isilon_hadoop_tools/blob/238bf718e6eb439a49aca8d8d5ae29df95b176a6/tests/test_identities.py#L7-L13

**Comment by tucked** [comment]
Closing and reopening to kick Travis...

**Comment by shirleylxie** [comment]
I missed to commit one file...

**Comment by shirleylxie** [comment]
Test result:
root@remotedev-sxie-rdu:/git/isilon_hadoop_tools# tox -e py36 -- --address 10.205.229.176  --password a
http://paste.west.isilon.com/247463

**Review by tucked — COMMENTED** [review]
Looks great! I've kicked off a full test run and will post the results Monday 😊

File: src/isilon_hadoop_tools/identities.py (hunk 2 above)
Comment (shirleylxie) [reply src/isilon_hadoop_tools/identities.py]: fixed.

File: src/isilon_hadoop_tools/_scripts.py [hunk 4]
```diff
@@ -33,7 +33,7 @@ def base_cli(parser=None):
     parser.add_argument(
         '--dist',
         help='the Hadoop distribution to be deployed',
-        choices=('cdh', 'hdp'),
+        choices=('cdh', 'hdp', 'cdp'),
```
[the remark this answers is not in the document]
Comment (shirleylxie) [reply src/isilon_hadoop_tools/_scripts.py]: fixed.

**Comment by shirleylxie** [comment]
Test result after fix:
http://paste.west.isilon.com/248229
http://paste.west.isilon.com/248230

**Review by tucked — APPROVED** [review]
Passing run against 8e3ee11: https://gist.github.com/tucked/ae2e3f11117cb748bca1060fa67cf6fb

**Pull request merged**